rxbindings/src/main/java/com/amplifyframework/rx/RxApiBinding.java

```diff
@@ -168,4 +173,83 @@
             CancelableBehaviors.StreamEmitter<String, T, ApiException> method) {
         return CancelableBehaviors.toObservable(method);
     }
+
+    /**
+     * A class that represents a subscription operation and exposes
+     * observables for consumers to listen to subscription data and
+     * status events.
+     * @param <T> The type representing the subscription data.
+     */
+    public static final class RxSubscriptionOperation<T> implements Cancelable {
+        private BehaviorSubject<ConnectionState> connectionStateSubject;
+        private Observable<T> subscriptionData;
+        private Cancelable amplifyOperation;
+        private String subscriptionId;
+
+        private Consumer<String> onConnected = new Consumer<String>() {
```

Comment:
Maybe do this as an inner class (FooConsumer extends Consumer), and initialize it inside the constructor body? S.t. `private final Consumer<String> onConnected;` is up with the other ordinary declarations?